region/norfair/west/Ice Beam Gate Room.json

```diff
@@ -180,57 +167,146 @@
       }
     },
     {
-      "id": 2,
       "link": [1, 1],
-      "name": "Crystal Flash",
+      "name": "Land in Door Frame",
+      "entranceCondition": {
+        "comeInNormally": {}
+      },
       "requires": [
-        "h_canCrystalFlash"
+        "canPrepareForNextRoom",
+        "canCarefulJump",
+        {"doorUnlockedAtNode": 1}
       ],
-      "flashSuitChecked": true
+      "unlocksDoors": [
+        {"types": ["super"], "requires": ["canInsaneJump"]},
+        {"types": ["missiles", "powerbomb"], "requires": ["never"]}
+      ],
+      "note": "Jump through the door and shoot it open as you enter, landing on the door frame to avoid falling."
     },
     {
-      "id": 3,
       "link": [1, 1],
-      "name": "Carry G-Mode Back Through",
+      "name": "Door Stuck, Open Door",
       "entranceCondition": {
-        "comeInWithGMode": {
-          "mode": "direct",
-          "morphed": false
-        }
+        "comeInWithDoorStuckSetup": {}
       },
-      "requires": [],
-      "exitCondition": {
-        "leaveWithGMode": {
-          "morphed": false
+      "requires": [
+        "canPrepareForNextRoom",
+        {"doorUnlockedAtNode": 1}
+      ],
+      "unlocksDoors": [
+        {"types": ["ammo"], "requires": []}
+      ]
+    },
+    {
+      "link": [1, 1],
+      "name": "Prepared Entry",
+      "entranceCondition": {
+        "comeInNormally": {}
+      },
+      "requires": [
+        "canPrepareForNextRoom",
```

Comment:
At this height you can react in time and don't need to be prepared.  But the springball option would need a movement tech too for placement.

Reply:
Added a careful jump to the spring ball option. You're right that it's really not that bad with this much space. I think its probably on the upper end of medium (assuming you dont know what room this will be). Do you think that is the right difficulty?

I kept the spring ball option free when coming in with mockball, as youre already morphed so it is trivial (you would also have the mockball tech turned on, so it's probably never a worry anyway).

Reply:
Feels too hard for Medium to me.  Hard doesn't have a great movement tech for placement but it could be like `canLateralMidAirMorph` since there is a canMidairMorph component.

Reply:
If you walk into this room without knowing what's coming, I don't think there's really enough time to react by morphing and bouncing, at the Hard difficulty level. That would seem more like Expert to think/react that quickly in a niche scenario like this. Morphing and bouncing is probably not very intuitive as a reaction anyway; I think most players would just try to crumble jump. To me a `canPrepareForNextRoom` requirement seems appropriate for the Spring Ball option. The `canCrumbleJump` option could be ok without `canPrepareForNextRoom`, since crumble jumping doesn't require as quick of a reaction.

Reply:
I agree with that.  The panic reaction wouldn't be to morph.